core/services/webhook/delegate.go

```diff
@@ -67,6 +67,9 @@ func (d *Delegate) BeforeJobDeleted(jb job.Job) {
 
 func (d *Delegate) ServicesForSpec(spec job.Job) ([]job.Service, error) {
 	// TODO: we need to fill these out manually, find a better fix
+	if spec.PipelineSpec == nil {
```

Comment:
Was this an accidental commit?

Reply:
not really, there is a panic case on the next line because `pec.PipelineSpec` can be nil

Reply:
It should never be `nil` but it sometimes happens during test cleanup because the job spawner loop keeps running. I'm trying to fix that here https://github.com/smartcontractkit/chainlink/pull/4994/commits/778754e1e1d97b9a0170e01e4c84cd0c28814b59

Reply:
I believe this check is not redundant. That protects from all possible nil pointer panics. Don't you think so?